script/run-league

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""
+A script to run a competition league specified in a compstate league file
+"""
+import sys
+import argparse
+from typing import TextIO
+from pathlib import Path
+
+import yaml
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+import match_utils  # isort:skip
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory containing the teams' robot code, as Zip archives "
+            "named for the teams' TLAs. This directory will also be used as the "
+            "root for storing the resulting logs and recordings."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        'league_file',
+        help=(
+            "The file containing a list of matches to run "
+            "as used by the compstate."
+        ),
+        type=argparse.FileType('r'),
+    )
```

Comment:
Reading just the league file is unfortunately not sufficient to correctly know the competitors who should be in a match. I would _strongly_ encourage that either the `srcomp` library or the related CLI package be used to get information about a compstate. Doing otherwise is likely to lead to bugs due to mismatches between how it reads the compstate (which it considers to be internal) and how your implementation does.

As an example, one way this is insufficient is that it doesn't account for teams who have dropped out since the start of the competition. They will still be present in the league file, but not actually expected in the match.

Using `srcomp` itself will be needed for the knockouts anyway as the matches aren't recorded in a file. We may as well therefore do that now and avoid the chances of complex and subtle issues.